### PR TITLE
DNS SDK changes for AutoRest

### PR DESCRIPTION
**Pull request by karravi-msft**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@karravi-msft__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (karravi). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/Dns/Dns.Tests/Helpers/RecordedDelegatingHandler.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+        public HttpMethod Method { get; private set; }
+
+        public Uri Uri { get; private set; }
+
+        public bool IsPassThrough { get; set; }
+
+        private int counter;
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, System.Threading.CancellationToken cancellationToken)
```
Comment (markcowl) [inline src/ResourceManagement/Dns/Dns.Tests/Helpers/RecordedDelegatingHandler.cs]: Fix line breaks

File: src/ResourceManagement/Dns/Dns.Tests/Helpers/ResourceGroupHelpers.cs [hunk 2]
```diff
@@ -74,28 +89,21 @@ public static ResourceGroupExtended CreateResourceGroup()
                     Location = location
                 });
 
-            return response.ResourceGroup;
+            return response;
         }
 
-        public static ZoneCreateOrUpdateResponse CreateZone(DnsManagementClient dnsClient, string zoneName, string location, ResourceGroupExtended resourceGroup)
+        public static Zone CreateZone(DnsManagementClient dnsClient, string zoneName, string location, ResourceGroup resourceGroup)
```
Comment (markcowl) [inline src/ResourceManagement/Dns/Dns.Tests/Helpers/ResourceGroupHelpers.cs]: Fix line breaks

File: src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/RecordSetScenarioTests.cs [hunk 3]
```diff
@@ -39,167 +41,165 @@ public class SingleRecordSetTestContext
...
+            testContext.ResourcesHandler = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+            testContext.DnsHandler = new RecordedDelegatingHandler { StatusCodeToReturn = HttpStatusCode.OK };
+            testContext.DnsClient = ResourceGroupHelper.GetDnsClient(context, testContext.DnsHandler);
+            var resourceManagementClient = ResourceGroupHelper.GetResourcesClient(context, testContext.ResourcesHandler);
+            testContext.ZoneName = TestUtilities.GenerateName("hydratestdnszone.com");
             testContext.RecordSetName = TestUtilities.GenerateName("hydratestdnsrec");
-            testContext.Location = ResourceGroupHelper.GetResourceLocation(ResourceGroupHelper.GetResourcesClient(), "microsoft.network/dnszones");
-            testContext.ResourceGroup = ResourceGroupHelper.CreateResourceGroup();
-            testContext.DnsClient = ResourceGroupHelper.GetDnsClient();
+            testContext.Location = ResourceGroupHelper.GetResourceLocation(resourceManagementClient, "microsoft.network/dnszones");
```
Comment (markcowl) [inline src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/RecordSetScenarioTests.cs]: Fix line breaks

File: src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/RecordSetScenarioTests.cs [hunk 4]
```diff
@@ -638,102 +535,112 @@ public void UpdateRecordSetPreconditionFailed()
...
                         testContext.ResourceGroup.Name,
                         testContext.ZoneName,
                         testContext.RecordSetName,
                         RecordType.CNAME,
-                        new RecordSetDeleteParameters { IfMatch = null });
+                        ifMatch: null,
+                        ifNoneMatch: null);
 
-                testContext.DnsClient.Zones.Delete(testContext.ResourceGroup.Name, testContext.ZoneName, new ZoneDeleteParameters());
+                testContext.DnsClient.Zones.Delete(testContext.ResourceGroup.Name, testContext.ZoneName, ifMatch: null, ifNoneMatch: null);
```
Comment (markcowl) [inline src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/RecordSetScenarioTests.cs]: Fix line breaks

File: src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/ZoneScenarioTests.cs [hunk 5]
```diff
@@ -17,160 +17,162 @@
...
-                assertZoneInvariants(getresponse.Zone);
-                Assert.Equal(1, getresponse.Zone.Tags.Count);
+                Assert.True(retrievedZone.NameServers != null && retrievedZone.NameServers.Any(nameServer => !string.IsNullOrWhiteSpace(nameServer)));
 
                 // Call Update on the object returned by Create (important distinction from Get below)
-                Zone createdZone = createResponse.Zone;
                 createdZone.Tags = new Dictionary<string, string> { { "tag1", "value1" }, { "tag2", "value2" } };
 
-                ZoneCreateOrUpdateResponse updateResponse = dnsClient.Zones.CreateOrUpdate(resourceGroup.Name, zoneName, new ZoneCreateOrUpdateParameters { Zone = createdZone });
+                var updateResponse = dnsClient.Zones.CreateOrUpdate(resourceGroup.Name, zoneName, ifMatch: null, ifNoneMatch: null, parameters: createdZone );
```
Comment (markcowl) [inline src/ResourceManagement/Dns/Dns.Tests/ScenarioTests/ZoneScenarioTests.cs]: line breks (a few in this file)

**Comment by markcowl** [comment]
@Karthikeyan global nit on tests, please make sure to break lines so they fit on the screen in github (~80 columns)  I've flagged a few instances.  Otherwise this looks good.

**Comment by karravi-msft** [comment]
@markcowl 
 The files had a higher line break setting even earlier.. Can we leave them as it is for now ?

I updated the pull request  with a couple of minor changes..can you please merge them too to main fork

**Pull request merged**